### PR TITLE
Blacklight 6

### PR DESCRIPTION
**Pull request by barmintor**
Done:
Subsites working
Repository sites working
Site working
CSV download working
Date Range Selector js working
Added a content availability facet to repo reading room catalogs

### CODE REVIEW

**Comment by barmintor** [comment]
[asset-reindex.csv.zip](https://github.com/cul/ldpd-dcv/files/4314792/asset-reindex.csv.zip)
Need to reindex these IDs after deploy.

File: app/controllers/repositories/catalog_controller.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
+module Repositories
+  class CatalogController < ::SubsitesController
+    include Dcv::MapDataController
+
+    before_action :set_map_data_json, only: [:map_search]
+    #before_action :set_map_data_json, only: [:index, :map_search]
```
Comment (elohanlon) [inline app/controllers/repositories/catalog_controller.rb]:
```suggestion
```

File: app/controllers/carnegie_controller.rb [hunk 2]
```diff
@@ -5,6 +5,8 @@ class CarnegieController < SubsitesController
   before_action :set_map_data_json, only: [:map_search]
   #before_action :set_map_data_json, only: [:index, :map_search]
```
Comment (elohanlon) [inline app/controllers/carnegie_controller.rb]:
```suggestion
```

File: app/helpers/dcv/children_helper_behavior.rb [hunk 3]
```diff
@@ -146,15 +146,16 @@ def structured_children_from_solr
...
+    response, docs = (defined? :controller) ? controller.search_results(_params) : search_results(_params)
     proxies = response['response']['docs']
     proxies = Hash[proxies.map {|proxy| [proxy['proxyFor_ssi'], proxy]}]
 
     _params[:q] = "{!join to=dc_identifier_ssim from=proxyFor_ssi}proxyIn_ssi:\"info:fedora/#{@document['id']}\""
     _params.delete(:fq)
 
-    response = Blacklight.solr.get 'select', params: _params
-    kids = response['response']['docs'].select { |doc| online_access_indicated?(doc) }
+    puts JSON.pretty_generate(_params)
```
Comment (elohanlon) [inline app/helpers/dcv/children_helper_behavior.rb]: Leftover `puts`.

Comment (barmintor) [reply app/helpers/dcv/children_helper_behavior.rb]: OH no! lol thanks

**Pull request merged**